common/src/main/java/com/wynntils/core/webapi/WebManager.java

```diff
@@ -231,12 +234,69 @@ public static boolean tryLoadItemList() {
                 .useCacheAsBackup()
                 .build());
 
-        handler.dispatch(false);
-
         // Check for success
         return isItemListLoaded();
     }
 
+    public static CompletableFuture<Boolean> tryLoadMap() {
+        if (apiUrls == null || !apiUrls.hasKey("AMainMap")) return CompletableFuture.completedFuture(false);
+
+        final File mapDirectory = new File(API_CACHE_ROOT, "map");
+
+        mapDirectory.mkdirs();
+
+        final CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+        handler.addAndDispatch(new RequestBuilder(apiUrls.get("AMainMap"), "main_map.info")
+                .cacheTo(new File(mapDirectory, "main-map.txt"))
+                .handleJson(json -> {
+                    JsonObject mapData = json.getAsJsonArray().get(0).getAsJsonObject();
```

Comment:
Since we only have a single map image, this works fine for now. But maybe add a comment describing this limitation?

Reply:
We should also have variable names for the files. I commented on https://github.com/Wynntils/WynntilsWebsite-API/pull/7 with an idea on how to do that. Which is trivial, really.

Reply:
The "obvious" solution is to name the files after the hash, I guess. But then we should really have some method of cleaning out old files as well. So if the files are named individually per segment, we need not bother about that part.

Reply:
We could also remove all files that did not match any hash in that directory

Reply:
Yes, that is a solution. Or we name the files after the coordinates: "-1331,33,-12,553.png" or so.

Reply:
Well, if we split the larger map file into smaller maps. Or wynncraft has another expansion. We are still stuck with leftover files

Reply:
Leftover files are really only a problem when files are renamed around, which should be rather infrequent. I feel it might be better to deal with that in a separate pr.